nav2_system_tests/src/recoveries/spin/spin_recovery_tester.cpp

```diff
@@ -192,6 +245,32 @@ bool SpinRecoveryTester::defaultSpinRecoveryTest(
   return true;
 }
 
+void SpinRecoveryTester::sendFakeCostmap(float angle)
```

Comment:
@SteveMacenski I think you asked this earlier -- this is the function that creates a fake costmap that is "populated" if the angle is greater than a particular value (pi / 2) to simulate the abort-ing of the spin recovery due to potential collision.  See `L264` .